### PR TITLE
Module for provisioning bloxone host in GCP

### PR DESCRIPTION
**Pull request by blodha-ib**

### CODE REVIEW

File: modules/bloxone_infra_host_gcp/userdata.tftpl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#cloud-config
+host_setup:
+  jointoken: ${join_token}
+  tags:
+  %{ for k, v in tags ~}
```
Comment (mathewab) [inline modules/bloxone_infra_host_gcp/userdata.tftpl:5]: why the `~` at the end here, I don't see it in AWS and Azure module

File: modules/bloxone_infra_host_gcp/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  value       = bloxone_infra_service.this
+  description = "The `bloxone_infra_service` objects for the instance. May be empty if no services were specified."
+}
+
+output "host" {
+  value       = data.bloxone_infra_hosts.this.results.0
+  description = "The `bloxone_infra_host` object for the instance"
+}
+
+output "gcp_instance" {
```
Comment (mathewab) [inline modules/bloxone_infra_host_gcp/outputs.tf:11]:
```suggestion
output "google_compute_instance" {
```

File: modules/bloxone_infra_host_gcp/main.tf [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+ * The service will be named `<service_type>_<host_display_name>`.
+ *
+/**
+ * ## Example Usage
+ *
+ * ```hcl
+ * module "bloxone_infra_host_gcp" {
+ *   source = "github.com/infobloxopen/terraform-provider-bloxone//modules/bloxone_infra_host_gcp"
+ *
+ *   name         = "bloxone-test-instance"
```
Comment (mathewab) [inline modules/bloxone_infra_host_gcp/main.tf:23]:
```suggestion
 *   name         = "bloxone-vm"
```

File: modules/bloxone_infra_host_gcp/main.tf [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+ *   project_id   = "gcp-engineering"
```
Comment (mathewab) [inline modules/bloxone_infra_host_gcp/main.tf:24]:
```suggestion
 *   project_id   = "my-project-id"
```

File: modules/bloxone_infra_host_gcp/variables.tf [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+variable "disk_type" {
+  description = "The type of the data disk."
+  type        = string
+  default     = "pd-standard"
+}
+
+variable "source_image" {
+  description = "The source image to use for the virtual machine."
+  type        = string
+}
```
Comment (mathewab) [inline modules/bloxone_infra_host_gcp/variables.tf:52]: A default would be useful here

File: modules/bloxone_infra_host_gcp/variables.tf [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+variable "gcp_instance_labels" {
+  description = "The labels to associate with the virtual machine"
```
Comment (mathewab) [inline modules/bloxone_infra_host_gcp/variables.tf:55]:
```suggestion
  description = "The labels to associate with the virtual machine. For tags to use in BloxOne resources, use `tags`."
```

File: modules/bloxone_infra_host_gcp/main.tf [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+      join_token = local.join_token
+      tags       = local.tags
+    })
+  }
+
+  service_account {
+    email = "var.service_account_email"
+    scopes = var.scopes
+  }
+}
```
Comment (mathewab) [inline modules/bloxone_infra_host_gcp/main.tf:117]: isn't this optional?

File: modules/bloxone_infra_host_gcp/main.tf [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+  name                = var.name
+  project             = var.project
+  zone                = var.zone
+  machine_type        = var.machine_type
+  labels              = var.gcp_instance_labels
+  deletion_protection = var.deletion_protection
+
+  boot_disk {
+    initialize_params {
+      image  = var.source_image
```
Comment (mathewab) [inline modules/bloxone_infra_host_gcp/main.tf:85]: you might also need a parameter for size. The image comes with only 59 GB, and the BloxOne documentation mentions to resize to 750 GB. 

https://docs.infoblox.com/space/BloxOneInfrastructure/350027851/GCP+Deployment#Resizing-the-disk-capacity

File: modules/bloxone_infra_host_gcp/userdata.tftpl (hunk 1 above)
Comment (blodha-ib) [reply modules/bloxone_infra_host_gcp/userdata.tftpl:5]: "~" removes the addition line that gets added after each "tag"  value

File: modules/bloxone_infra_host_gcp/outputs.tf (hunk 2 above)
Comment (blodha-ib) [reply modules/bloxone_infra_host_gcp/outputs.tf]: updated

File: modules/bloxone_infra_host_gcp/main.tf (hunk 3 above)
Comment (blodha-ib) [reply modules/bloxone_infra_host_gcp/main.tf]: updated

File: modules/bloxone_infra_host_gcp/main.tf (hunk 4 above)
Comment (blodha-ib) [reply modules/bloxone_infra_host_gcp/main.tf]: Removed some of the optional items that can be inherited from provider configuration.

File: modules/bloxone_infra_host_gcp/variables.tf (hunk 5 above)
Comment (blodha-ib) [reply modules/bloxone_infra_host_gcp/variables.tf:47]: added variable and default value ( with validation )

File: modules/bloxone_infra_host_gcp/variables.tf (hunk 6 above)
Comment (blodha-ib) [reply modules/bloxone_infra_host_gcp/variables.tf]: updated

File: modules/bloxone_infra_host_gcp/main.tf (hunk 7 above)
Comment (blodha-ib) [reply modules/bloxone_infra_host_gcp/main.tf]: Yes, this optional. Updated the code.

File: modules/bloxone_infra_host_gcp/main.tf (hunk 8 above)
Comment (blodha-ib) [reply modules/bloxone_infra_host_gcp/main.tf:80]: added variable ( with default value and validation )

File: modules/bloxone_infra_host_gcp/README.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+| <a name="input_source_image"></a> [source\_image](#input\_source\_image) | The source image to use for the virtual machine. | `string` | n/a | yes |
+| <a name="input_tags"></a> [tags](#input\_tags) | The tags to use for the BloxOne Host. | `map(string)` | `{}` | no |
+| <a name="input_timeouts"></a> [timeouts](#input\_timeouts) | The timeouts to use for the BloxOne Host. The timeout value is a string that can be parsed as a duration consisting of numbers and unit suffixes, such as "30s" or "2h45m". Valid time units are "s" (seconds), "m" (minutes), "h" (hours). If not provided, the default timeouts will be used. | <pre>object({<br>    create = string<br>    update = string<br>    read   = string<br>  })</pre> | `null` | no |
+
+## Outputs
+
+| Name | Description |
+|------|-------------|
+| <a name="output_gcp_compute_instance"></a> [gcp\_compute\_instance](#output\_gcp\_compute\_instance) | The GCP instance object for the instance |
```
Comment (mathewab) [inline modules/bloxone_infra_host_gcp/README.md:100]: need to regenerate docs?

**Review by mathewab — APPROVED** [review]

**Pull request merged**